File: frontend/src/types/fxPl.ts
export interface FxPlBreakdown {
  code: string
  stock_name: string
  trade_currency: string
  base_currency: string
  total_pl_base: number
  stock_pl_base: number
  fx_pl_base: number
  cost_at_purchase: number
  cost_at_current: number
  fx_rate_buy: number
  fx_rate_current: number
  hold_period_days: number
}

export interface FxPlSummary {
  total_pl_base: number
  total_stock_pl: number
  total_fx_pl: number
  fx_impact_ratio: number
  positions_with_fx_impact: number
  breakdowns: FxPlBreakdown[]
}

export interface CurrencyExposure {
  currency: string
  market_value_in_base: number
  weight_pct: number
  unrealized_fx_pl: number
  position_count: number
}

File: backend/app/models/fx_pl.py
from pydantic import BaseModel


class FxPlBreakdown(BaseModel):
    code: str
    stock_name: str
    trade_currency: str
    base_currency: str
    total_pl_base: float
    stock_pl_base: float
    fx_pl_base: float
    cost_at_purchase: float
    cost_at_current: float
    fx_rate_buy: float
    fx_rate_current: float
    hold_period_days: int


class FxPlSummary(BaseModel):
    total_pl_base: float
    total_stock_pl: float
    total_fx_pl: float
    fx_impact_ratio: float
    positions_with_fx_impact: int
    breakdowns: list[FxPlBreakdown]


class CurrencyExposure(BaseModel):
    currency: str
    market_value_in_base: float
    weight_pct: float
    unrealized_fx_pl: float
    position_count: int
